File: ide-test/src/test/java/org/example/BuildService.java
package org.example;

import org.example.remoting.Remote;
import org.example.shared.Ref;

@Remote("org.example.app.BuildService")
interface BuildService {
    BuildResult build(Ref projectRef, String args);
}

@Remote("org.example.app.BuildResult")
interface BuildResult {
    String getArgs();
}

File: ide-test/src/test/java/org/example/BuildService.kt
package org.example

import org.example.remoting.Remote
import org.example.remoting.sdk.Project

@Remote("org.example.app.BuildService")
internal interface BuildService {
    fun build(projectRef: Project?, args: String?): BuildResult
}

@Remote("org.example.app.BuildResult")
internal interface BuildResult {
    fun getArgs(): String
}